plugin/pkg/admission/resourcequota/admission.go

```diff
@@ -20,34 +20,43 @@ import (
 	"fmt"
 	"io"
 	"math/rand"
+	"strings"
 	"time"
 
 	"k8s.io/kubernetes/pkg/admission"
 	"k8s.io/kubernetes/pkg/api"
-	"k8s.io/kubernetes/pkg/api/resource"
-	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/api/errors"
 	"k8s.io/kubernetes/pkg/client/cache"
 	client "k8s.io/kubernetes/pkg/client/unversioned"
-	resourcequotacontroller "k8s.io/kubernetes/pkg/controller/resourcequota"
+	"k8s.io/kubernetes/pkg/quota"
+	"k8s.io/kubernetes/pkg/quota/install"
 	"k8s.io/kubernetes/pkg/runtime"
-	utilerrors "k8s.io/kubernetes/pkg/util/errors"
 	"k8s.io/kubernetes/pkg/watch"
 )
 
 func init() {
-	admission.RegisterPlugin("ResourceQuota", func(client client.Interface, config io.Reader) (admission.Interface, error) {
-		return NewResourceQuota(client), nil
-	})
+	admission.RegisterPlugin("ResourceQuota",
+		func(client client.Interface, config io.Reader) (admission.Interface, error) {
+			registry := install.NewRegistry(client)
```

Comment:
@miminar - downstream should instantiate the plug-in with its own registry.  see similar pattern here for how downstream is injecting values into admission handlers: https://github.com/openshift/origin/blob/master/pkg/cmd/server/origin/master_config.go#L181